### PR TITLE
Add DVT flag

### PR DESCRIPTION
**Pull request by mcdee**

### CODE REVIEW

File: http/beaconblockproposal.go [hunk 1]
```diff
@@ -87,11 +87,15 @@ func (s *Service) beaconBlockProposalV2(ctx context.Context, slot phase0.Slot, r
 		if resp.Data.Slot != slot {
 			return nil, errors.New("beacon block proposal not for requested slot")
 		}
-		if !bytes.Equal(resp.Data.Body.RANDAOReveal[:], randaoReveal[:]) {
-			return nil, errors.New("beacon block proposal has incorrect RANDAO reveal")
-		}
-		if !bytes.Equal(resp.Data.Body.Graffiti[:], graffiti) {
-			return nil, errors.New("beacon block proposal has incorrect graffiti")
+		// Only check the RANDAO reveal and agraffiti if we are not connected to DVT middleware,
```
Comment (dB2510) [inline http/beaconblockproposal.go:90]:
```suggestion
		// Only check the RANDAO reveal and graffiti if we are not connected to DVT middleware,
```

File: http/beaconblockproposal.go [hunk 2]
```diff
@@ -103,11 +107,15 @@ func (s *Service) beaconBlockProposalV2(ctx context.Context, slot phase0.Slot, r
...
```
Comment (dB2510) [inline http/beaconblockproposal.go:110]:
```suggestion
		// Only check the RANDAO reveal and graffiti if we are not connected to DVT middleware,
```

File: http/beaconblockproposal.go [hunk 3]
```diff
@@ -119,11 +127,15 @@ func (s *Service) beaconBlockProposalV2(ctx context.Context, slot phase0.Slot, r
...
```
Comment (dB2510) [inline http/beaconblockproposal.go:130]:
```suggestion
		// Only check the RANDAO reveal and graffiti if we are not connected to DVT middleware,
```

File: http/beaconblockproposal.go [hunk 4]
```diff
@@ -135,11 +147,15 @@ func (s *Service) beaconBlockProposalV2(ctx context.Context, slot phase0.Slot, r
...
```
Comment (dB2510) [inline http/beaconblockproposal.go:150]:
```suggestion
		// Only check the RANDAO reveal and graffiti if we are not connected to DVT middleware,
```

File: http/beaconblockproposal.go [hunk 5]
```diff
@@ -169,8 +185,12 @@ func (s *Service) beaconBlockProposalV1(ctx context.Context, slot phase0.Slot, r
...
-	if !bytes.Equal(resp.Data.Body.RANDAOReveal[:], randaoReveal[:]) {
-		return nil, errors.New("beacon block proposal has incorrect RANDAO reveal")
+	// Only check the RANDAO reveal if we are not connected to DVT middleware, as that
+	// will return a full signature rather than the element signature we passed it.
+	if !s.connectedToDVTMiddleware {
+		if !bytes.Equal(resp.Data.Body.RANDAOReveal[:], randaoReveal[:]) {
+			return nil, errors.New("beacon block proposal has incorrect RANDAO reveal")
+		}
 	}
 	if !bytes.Equal(resp.Data.Body.Graffiti[:], graffiti) {
```
Comment (dB2510) [inline http/beaconblockproposal.go:195]: can we also include this graffiti check inside the above if block?

File: http/blindedbeaconblockproposal.go [hunk 6]
```diff
@@ -76,11 +76,15 @@ func (s *Service) blindedBeaconBlockProposal(ctx context.Context, slot phase0.Sl
 		if resp.Data.Slot != slot {
 			return nil, errors.New("blinded beacon block proposal not for requested slot")
 		}
-		if !bytes.Equal(resp.Data.Body.RANDAOReveal[:], randaoReveal[:]) {
-			return nil, errors.New("blinded beacon block proposal has incorrect RANDAO reveal")
-		}
-		if !bytes.Equal(resp.Data.Body.Graffiti[:], graffiti) {
-			return nil, errors.New("blinded beacon block proposal has incorrect graffiti")
+		// Only check the RANDAO reveal and agraffiti if we are not connected to DVT middleware,
```
Comment (dB2510) [inline http/blindedbeaconblockproposal.go:79]:
```suggestion
		// Only check the RANDAO reveal and graffiti if we are not connected to DVT middleware,
```

File: http/blindedbeaconblockproposal.go [hunk 7]
```diff
@@ -92,11 +96,15 @@ func (s *Service) blindedBeaconBlockProposal(ctx context.Context, slot phase0.Sl
...
```
Comment (dB2510) [inline http/blindedbeaconblockproposal.go:99]:
```suggestion
		// Only check the RANDAO reveal and graffiti if we are not connected to DVT middleware,
```

**Review by dB2510 — APPROVED** [review]
LGTM 🚀

File: http/beaconblockproposal.go (hunk 5 above)
Comment (mcdee) [reply http/beaconblockproposal.go]: Done.

**Pull request merged**